File: SmartClinic-Server/Models/Doctor.js
import prisma from "../lib/prisma.js";

export class Doctor {

    static async findAll() {
        return await prisma.doctor.findMany();
      }

      static async findUniqueDoctor(email) {
        return prisma.doctor.findUnique({
          where: {
            email,
          },
        });
      }

      static async allWithSchedules() {
        return await prisma.doctor.findMany({
            select: {
              id: true,
              name: true,
              email: true,
              yearsExperience: true,
              profilePicture: true,
              schedules: {
                select: {
                  id: true,
                  startTime: true,
                  endTime: true,
                  slotDuration: true,
                  offdays: true,
                },
              },
            },
          });
      }

      static async CreateDoctor(data) {
        const { name, email, password, yearsExperience, profilePicture } = data;
        return await prisma.doctor.create({
          data: {
            name,
            email,
            password,
            yearsExperience,
            profilePicture,
          },
        });
      }
} 

File: SmartClinic-Mobile/patient_app/lib/models/doctor.dart
class Schedule {
  final int id;
  final String startTime;
  final String endTime;
  final int slotDuration;
  final List<String> offdays;

  Schedule({
    required this.id,
    required this.startTime,
    required this.endTime,
    required this.slotDuration,
    required this.offdays,
  });

  factory Schedule.fromJson(Map<String, dynamic> json) {
    return Schedule(
      id: json['id'],
      startTime: json['startTime'].replaceAll('"', ''),
      endTime: json['endTime'].replaceAll('"', ''),
      slotDuration: json['slotDuration'],
      offdays: List<String>.from(json['offdays']),
    );
  }
}

class Doctor {
  final int id;
  final String name;
  final String email;
  final int yearsExperience;
  final String profilePicture;
  final List<Schedule> schedules;

  Doctor({
    required this.id,
    required this.name,
    required this.email,
    required this.yearsExperience,
    required this.profilePicture,
    required this.schedules,
  });

  factory Doctor.fromJson(Map<String, dynamic> json) {
    final schedulesJson = json['schedules'];

    return Doctor(
      id: json['id'],
      name: json['name'],
      email: json['email'],
      yearsExperience: json['yearsExperience'],
      profilePicture: json['profilePicture'],
      schedules:
          schedulesJson is List
              ? schedulesJson.map((e) => Schedule.fromJson(e)).toList()
              : [],
    );
  }
}
